llvm/include/llvm/CodeGen/SelectionDAGNodes.h

```diff
@@ -1678,6 +1678,7 @@ class ConstantSDNode : public SDNode {
       : SDNode(isTarget ? ISD::TargetConstant : ISD::Constant, 0, DebugLoc(),
                VTs),
         Value(val) {
+    assert(!isa<VectorType>(val->getType()) && "Unexpected vector type!");
```

Comment:
This is just a thought, but is it worth tying these asserts down even further, i.e.

  assert(val->getType()->isIntegerTy() && "Unexpected type!");

and similarly for the FP node? At the moment, these nodes still won't assert for any other unusual types.

Reply:
That's handled as part of `ConstantInt` construction and thus here we can safely assume `val` has a valid type for its class.